lib/router/route-info.ts

```diff
@@ -375,12 +370,12 @@ export default class InternalRouteInfo<T extends Route> {
     });
   }
 
-  private checkForAbort<T>(shouldContinue: Continuation, value: T) {
-    return Promise.resolve(shouldContinue()).then(function () {
-      // We don't care about shouldContinue's resolve value;
-      // pass along the original value passed to this fn.
-      return value;
-    }, null);
+  private checkForAbort<U>(transition: InternalTransition<T>, value: U) {
```

Comment:
actually, maybe something like the following signature should tighten things up nicely, as it will now be able to perform inference through `checkForAbort`

```ts
interface AbortableTransition<T extends boolean, R extends Route> extends InternalTransition<R> {
  isAborted: T;
}

function checkForAbort<U, T extends AbortableTransition<I, R>, R extends Route, I extends boolean>(
  transition: T,
  value: U
): I extends true ? never : U;
function checkForAbort<U, T extends AbortableTransition<I, R>, R extends Route, I extends boolean>(
  transition: T,
  value: U
): never | U {
  if (transition.isAborted) {
    throw new Error('Transition aborted');
  }

  return value;
}
```

That ended up being a little messier then expected, but now it's written I may as well add it.

Reply:
Nice, thanks for doing that @stefanpenner!